sdk/cosmos/azure-cosmos-benchmark/src/main/java/com/azure/cosmos/benchmark/linkedin/ResourceManagerImpl.java

```diff
@@ -0,0 +1,111 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.cosmos.benchmark.linkedin;
+
+import com.azure.cosmos.CosmosAsyncClient;
+import com.azure.cosmos.CosmosAsyncContainer;
+import com.azure.cosmos.CosmosAsyncDatabase;
+import com.azure.cosmos.CosmosException;
+import com.azure.cosmos.benchmark.Configuration;
+import com.azure.cosmos.models.CosmosContainerProperties;
+import com.azure.cosmos.models.ThroughputProperties;
+import com.google.common.base.Preconditions;
+import java.time.Duration;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static com.azure.cosmos.benchmark.linkedin.impl.Constants.PARTITION_KEY_PATH;
+import static com.azure.cosmos.models.ThroughputProperties.createManualThroughput;
+
+
+public class ResourceManagerImpl implements ResourceManager {
+    private static final Logger LOGGER = LoggerFactory.getLogger(ResourceManagerImpl.class);
+    private static final Duration RESOURCE_CRUD_WAIT_TIME = Duration.ofSeconds(30);
+
+    private final Configuration _configuration;
+    private final CosmosAsyncClient _client;
+
+    public ResourceManagerImpl(final Configuration configuration, final CosmosAsyncClient client) {
+        Preconditions.checkNotNull(configuration,
+            "The Workload configuration defining the parameters can not be null");
+        Preconditions.checkNotNull(client, "Need a non-null client for "
+            + "setting up the Database and containers for the test");
+        _configuration = configuration;
+        _client = client;
+    }
+
+    @Override
+    public void createDatabase() throws CosmosException {
+        try {
+            LOGGER.info("Creating database {} for the ctl workload", _configuration.getDatabaseId());
+            _client.createDatabaseIfNotExists(_configuration.getDatabaseId())
+                .block(RESOURCE_CRUD_WAIT_TIME);
+        } catch (CosmosException e) {
+            LOGGER.error("Exception while creating database {}", _configuration.getDatabaseId(), e);
+            throw e;
+        }
+
+        deleteExistingContainers();
+    }
+
+    @Override
+    public void createContainer() throws CosmosException {
+        final String containerName = _configuration.getCollectionId();
+        final CosmosAsyncDatabase database = _client.getDatabase(_configuration.getDatabaseId());
+        final ThroughputProperties containerThroughput = createManualThroughput(_configuration.getThroughput());
```

Comment:
Please take throughput to database level, that's what we are doing on sdk workload, and also if you need to add new container , you need not to configure throughput again.

Reply:
Updated in the latest revision